tests/test_integration_layout.py

```diff
@@ -246,32 +247,43 @@ def compose(self) -> ComposeResult:
         # explicit hiding of the overflow: no scrollbars either
         [Size(30, 30), "overflow: hidden", Size(20, 20)],
         # scrollbar for both directions
-        [Size(30, 30), "overflow: auto", Size(19, 19)],
+        [
+            Size(30, 30),
+            "overflow: auto",
+            Size(
+                20 - SCROLL_V_SIZE,
+                20 - SCROLL_H_SIZE,
+            ),
+        ],
         # horizontal scrollbar
-        [Size(30, 30), "overflow-x: auto", Size(20, 19)],
+        [Size(30, 30), "overflow-x: auto", Size(20, 20 - SCROLL_H_SIZE)],
         # vertical scrollbar
-        [Size(30, 30), "overflow-y: auto", Size(19, 20)],
+        [Size(30, 30), "overflow-y: auto", Size(20 - SCROLL_V_SIZE, 20)],
         # scrollbar for both directions, custom scrollbar size
-        [Size(30, 30), ("overflow: auto", "scrollbar-size: 3 5"), Size(15, 17)],
+        [Size(30, 30), ("overflow: auto", "scrollbar-size: 3 5"), Size(20 - 5, 20 - 3)],
         # scrollbar for both directions, custom vertical scrollbar size
-        [Size(30, 30), ("overflow: auto", "scrollbar-size-vertical: 3"), Size(17, 19)],
+        [
+            Size(30, 30),
+            ("overflow: auto", "scrollbar-size-vertical: 3"),
+            Size(20 - 3, 20 - SCROLL_H_SIZE),
+        ],
         # scrollbar for both directions, custom horizontal scrollbar size
         [
             Size(30, 30),
             ("overflow: auto", "scrollbar-size-horizontal: 3"),
-            Size(19, 17),
+            Size(20 - SCROLL_V_SIZE, 20 - 3),
         ],
         # scrollbar needed only vertically, custom scrollbar size
         [
             Size(20, 30),
             ("overflow: auto", "scrollbar-size: 3 3"),
-            Size(17, 20),
+            Size(20 - 3, 20),
         ],
         # scrollbar needed only horizontally, custom scrollbar size
         [
             Size(30, 20),
             ("overflow: auto", "scrollbar-size: 3 3"),
-            Size(20, 17),
+            Size(20, 20 - 3),
```

Comment:
While I was there... `20 - 3` might be a bit clearer regarding what this "expected size" number is: it's the size of the container (20, as explained at the beginning of the test' parameters list), minus the custom scrollbar size we define on the line above :slightly_smiling_face: